cob_cartesian_controller/src/simple_cartesian_interface/simple_cartesian_interface.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/python
+import rospy
+import threading
+import actionlib
+
+from tf.transformations import *
+from geometry_msgs.msg import Pose
+from cob_cartesian_controller.msg import CartesianControllerAction, CartesianControllerGoal
+from cob_cartesian_controller.msg import Profile
+
+
+_transform_listener = None
+_transform_listener_ock = threading.Lock()
+
+_action_client = None
+_action_client_lock = threading.Lock()
+
+def move_lin(pose_goal, frame_id, profile, rotate_only=False):
+    client = actionlib.SimpleActionClient('cartesian_trajectory_action', CartesianControllerAction)
```

Comment:
Create class that provides a member for the SimpleActionClient instead of creating one with each call. 

Additionally there is no need to wait for the server in each move_lin call.